### DISCUSSION TITLE
Clarify what concurrency really means and its syntax

### DISCUSSION THREAD

**Discussion by JJ** [community/community#25782] [Actions]
<p>Checking the documents on <a href="https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#concurrency" rel="noopener nofollow ugc">concurrency</a>, I couldn’t quite understand how to apply them to specific use cases. Suppose that we have three workflows, A, B and C. B and C take a long time, A is fast, but it’s quite convenient to run them at the same time, that is, concurrently.<br>
However, I want B to stop if A fails, and also C to stop if B fails. So how can I do this?<br>
My hunch would be to have B belonging to two concurrency groups, and then use <code>cancel-in-progress</code> (which I understand does not prevent them running at the same time). But alas, there does not seem to be a syntax that allows a workflow to belong to two groups. Is that the case? Is there any workaround?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Basically, if you want jobs to run in parallel, <em>don’t</em> use the <code>concurrency</code> setting. It’s for limiting simultaneous runs from different events.</p>
<p>By default workflows and jobs started from the same event run in parallel as long as enough runners are available. I’m not aware of any way to cancel simultaneous jobs if one of them fails outside a <code>matrix</code> strategy, unfortunately.</p>

**Comment by jsoref** [comment]
<p>You can use the API to cancel jobs, so you could use an extra step that runs on failure() to have it shoot other things.</p>
<p>The <code>gh</code> client can do it:</p><aside class="onebox githubpullrequest" data-onebox-src="https://github.com/cli/cli/pull/3833">
  <header class="source">

      <a href="https://github.com/cli/cli/pull/3833" target="_blank" rel="noopener nofollow ugc">github.com/cli/cli</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Pull Request">
    <svg width="60" height="60" class="github-icon" viewbox="0 0 12 16" aria-hidden="true"><path d="M11 11.28V5c-.03-.78-.34-1.47-.94-2.06C9.46 2.35 8.78 2.03 8 2H7V0L4 3l3 3V4h1c.27.02.48.11.69.31.21.2.3.42.31.69v6.28A1.993 1.993 0 0 0 10 15a1.993 1.993 0 0 0 1-3.72zm-1 2.92c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zM4 3c0-1.11-.89-2-2-2a1.993 1.993 0 0 0-1 3.72v6.56A1.993 1.993 0 0 0 2 15a1.993 1.993 0 0 0 1-3.72V4.72c.59-.34 1-.98 1-1.72zm-.8 10c0 .66-.55 1.2-1.2 1.2-.65 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/cli/cli/pull/3833" target="_blank" rel="noopener nofollow ugc">Add `run cancel` command</a>
    </h4>

    <div class="branches">
      <code>cli:trunk</code> ← <code>cristiand391:gh-run-cancel</code>
    </div>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2021-06-14" data-time="12:05:57" data-timezone="UTC">12:05PM - 14 Jun 21 UTC</span>
      </div>

      <div class="user">
        <a href="https://github.com/cristiand391" target="_blank" rel="noopener nofollow ugc">
          <img alt="cristiand391" src="https://user-images.githubusercontent.com/2119212/181093255-e935cc9e-8551-4ee2-b7ca-64830b46f16e.png" class="onebox-avatar-inline" width="20" height="20">
          cristiand391
        </a>
      </div>

      <div class="lines" title="8 commits changed 3 files with 356 additions and 0 deletions">
        <a href="https://github.com/cli/cli/pull/3833/files" target="_blank" rel="noopener nofollow ugc">
          <span class="added">+356</span>
          <span class="removed">-0</span>
        </a>
      </div>
    </div>
  </div>
</div>

  <div class="github-row">
    <p class="github-body-container">This PR implements a new command for canceling a workflow run that's in progress<span class="show-more-container"><a href="https://github.com/cli/cli/pull/3833" target="_blank" rel="noopener nofollow ugc" class="show-more">…</a></span><span class="excerpt hidden">.

Closes #3775</span></p>
  </div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by airtower-luna** [comment]
<p>Good point, I should’ve written that there’s nothing built into the workflow syntax for that purpose. API works. :slightly_smiling_face:</p>
